src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
+): GroupsMap {
+  const result: GroupsMap = {};
+
+  for (const item of items) {
+    if (result.hasOwnProperty(item[key])) {
```

Comment:
Using `hasOwnProperty` directly from the result object can lead to unexpected behavior if the object has a property with the same name as a method from `Object.prototype`. It's safer to use `Object.prototype.hasOwnProperty.call(result, item[key])`.